src/tesseract.rs

```diff
@@ -74,10 +74,40 @@ impl TessApi {
         }
     }
 
+    /// Provide an image for Tesseract to recognize.
+    /// 
+    /// set_image clears all recognition results, and sets the rectangle to the full image, so it
+    /// may be followed immediately by a `[Self::get_utf8_text]`, and it will automatically perform
+    /// recognition. 
     pub fn set_image(&mut self, img: &leptonica::Pix) {
+        // "Tesseract takes its own copy of the image, so it need not persist until after Recognize"
         unsafe { capi::TessBaseAPISetImage2(self.raw, img.raw as *mut capi::Pix) }
     }
 
+    /// Get the dimensions of the currently loaded image, or None if no image is loaded.
+    /// 
+    /// # Example
+    /// ```rust
+    /// let path = std::path::Path::new("tests/di.png");
+    /// let img = leptess::leptonica::pix_read(&path).unwrap();
+    /// 
+    /// let mut tes = leptess::tesseract::TessApi::new(Some("tests/tessdata"), "eng").unwrap();
+    /// tes.set_image(&img);
+    /// 
+    /// assert_eq!(tes.get_image_dimensions(), Some((442, 852)));
+    /// ```
+    pub fn get_image_dimensions(&self) -> Option<(u32, u32)> {
+        // - is TessBaseAPIGetInputImage the correct function to call? it's not documented at all
```

Comment:
> is TessBaseAPIGetInputImage the correct function to call? it's not documented at all

I think it is probably correct.

I agree, the documentation leaves a lot to be desired:
https://tesseract-ocr.github.io/tessapi/3.x/a00613.html#ab0cb619ceec1ca8872621399f8238d58

But as the name suggests it does what we want, and the tests pass, I'm confident it is the correct function.

>    are .w and .h the correct attributes to check?

This is better documented https://tpgit.github.io/Leptonica/struct_pix.html

We can see that `w` is referenced by `pixGetWidth()`, and `h` is referenced by `pixGetHeight()`. I think they are width and height which is the image dimensions.

I think you're doing this correctly 🙂

My preference would be not to keep this comment in the code

Reply:
That's fair, I'll remove